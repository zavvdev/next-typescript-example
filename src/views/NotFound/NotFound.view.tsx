import { useTranslation } from 'next-i18next';
import Link from 'next/link';
import I18N_NAMESPACES from 'src/constants/i18nNamespaces';
import { GENERAL_ROUTES } from 'src/constants/routes';
import classes from 'src/views/NotFound/NotFound.module.scss';

const NotFoundView = () => {
  const { t } = useTranslation(I18N_NAMESPACES.notFound);

  return (
    <div className={classes.root}>
      <span>{t('code')}</span>
      <h1>{t('text')}</h1>
      <Link href={GENERAL_ROUTES.home}>
        <a className={classes.btn}>
          {t('buttons.home')}
        </a>
      </Link>
    </div>
  );
};

export default NotFoundView;
