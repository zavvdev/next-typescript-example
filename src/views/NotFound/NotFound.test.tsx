import { screen } from '@testing-library/react';
import render from 'src/tests/utils/render';
import NotFound from 'src/views/NotFound/NotFound.view';

describe('<NotFound />', () => {
  test('should render NotFound view (snapshot)', () => {
    const { asFragment } = render(<NotFound />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should have http code', () => {
    render(<NotFound />);
    expect(screen.getByText('code')).toBeInTheDocument();
  });

  test('should have a text', () => {
    render(<NotFound />);
    expect(screen.getByText('text')).toBeInTheDocument();
  });

  test('should have a Home button', () => {
    render(<NotFound />);
    expect(screen.getByText('buttons.home')).toBeInTheDocument();
  });
});
