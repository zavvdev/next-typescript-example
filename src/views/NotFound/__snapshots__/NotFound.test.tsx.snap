// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should render NotFound view (snapshot) 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <span>
      code
    </span>
    <h1>
      text
    </h1>
    <a
      class="btn"
      href="/"
    >
      buttons.home
    </a>
  </div>
</DocumentFragment>
`;
